### ISSUE TITLE
UnusedIO improvements

### ISSUE THREAD

**Issue by NeQuissimus** [typelevel/typelevel-scalafix#14]
I am just running through a few test scenarios :)

Detected:
```scala
  def test(): IO[Long] = for {
    a <- IO.pure(42L)
    _ = IO.println("hello")
  } yield a
```

Not detected: 
```scala
  def test(): IO[Long] = for {
    a <- IO.pure(42L)
    _ = IO { println("hello") }
  } yield a
```

Not detected:
```scala
  def test(): Long = {
    IO { println("hello") }
    42
  }
```

So far the rule definitely helps with common cases of "forgotten IO" :)

Just leaving these here as potential test cases for future improvements.

**Comment by DavidGregory084** [comment]
@NeQuissimus this will be fixed in 0.1.1, but please keep them coming if you find more!

**Comment by NeQuissimus** [comment]
Thank you, I will be sure to provide minimal repros whenever I run into anything :) Thanks for the quick fix
